### PR TITLE
[dynamo bug burndown] update tensor creation to support sequences of tensors

### PR DESCRIPTION
**Pull request by ColinPeppler**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #120872

Fixes https://github.com/pytorch/pytorch/issues/120645

`_internal_new_from_data` calls `_recursive_build`, but we run into an error such as the cases.
```
Failed running call_function <function tensor at 0xDEADBEEF>:
scalar_tensor(): argument (position 1) must be Number, not FakeTensor

# e.g. cases
1. [FakeTensor(..., size=(20, 1), dtype=torch.float64), ..., FakeTensor(..., size=(20, 1), dtype=torch.float64)]
- Here, we call _recursive_build(sizes=[4] ...) which hits the base case `if dim == ndim:` in the 2nd level of recursion.
- So, we try to return `scalar_tensor(FakeTensor)`
2. [[(FakeTensor(..., size=(1,), dtype=torch.int64), FakeTensor(..., size=(), dtype=torch.int64)]]

# site note: when can size = ()? Probably from scalar_tensor.
>>> torch.scalar_tensor(1).shape
torch.Size([])
```

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

**Comment by ColinPeppler** [comment]
TEsts pass :) need to run lintrunner + enable some more tests expected to fail

File: torch/_refs/__init__.py [hunk 1]
```diff
@@ -6332,21 +6332,15 @@ def _infer_scalar_type(obj):
 
 # Analogous to recursive_store
 # xref: recursive_store in torch/csrc/utils/tensor_new.cpp
-def _recursive_build(sizes, dim, scalarType, obj):
-    ndim = len(sizes)
-    assert dim <= ndim
-    if dim == ndim:
+def _recursive_build(scalarType, obj):
+    if isinstance(obj, Tensor) and obj.ndim <= 1:
+        obj = obj.item()
```
Comment (ezyang) [inline torch/_refs/__init__.py:6337]: No need to go through item(), just .to(scalarType, copy=True)

**Review by ezyang — APPROVED** [review]
nice, thanks

File: torch/_refs/__init__.py (hunk 1 above)
Comment (ColinPeppler) [reply torch/_refs/__init__.py:6337]: seems like `torch.stack` expects its entries to have the same size.

```
# tried this
-        obj = obj.item()
+        return obj.to(scalarType, copy=True)

# saw this
stack expects each tensor to be equal size, but got torch.Size([1]) at entry and torch.Size([]) at entry 1
```

**Pull request closed without merging**